src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/HeadLivenessValidator.kt

```diff
@@ -0,0 +1,44 @@
+package io.emeraldpay.dshackle.upstream.ethereum
+
+import io.emeraldpay.dshackle.upstream.Head
+import org.springframework.scheduling.concurrent.CustomizableThreadFactory
+import reactor.core.publisher.Flux
+import reactor.core.scheduler.Schedulers
+import java.time.Duration
+import java.util.concurrent.Executors
+
+class HeadLivenessValidator(
+    val head: Head,
+    val expectedBlockTime: Duration
+) {
+
+    val scheduler =
+        Schedulers.fromExecutor(Executors.newCachedThreadPool(CustomizableThreadFactory("head-liveness-validator")))
```

Comment:
It is better to use injected scheduler, one for all upstreams (like headScheduler).
Other option - Schedulers.parallel()
Last option - set scheduler outside this class where other schedulers available